lib/GroupBackend.php

```diff
@@ -7,6 +7,7 @@
 
 namespace OCA\User_SAML;
 
+use OC\User\LazyUser;
```

Comment:
This is problematic, it is not part of the API. It is also bad that  the internal backend makes use of it, while there is no chance for external ones. 

Would it be sufficient to implement `IProvideEnabledStateBackend` instead? Then the user management abstraction would create the LazyUser objects.